### PR TITLE
check for csrf token in header

### PR DESCRIPTION
**Pull request by phrix32**

### CODE REVIEW

File: application/service.go [hunk 1]
```diff
@@ -16,8 +16,10 @@ import (
 )
 
 const (
-	// TokenName is used to define form parameter name.
-	TokenName = "csrftoken"
+	// FormTokenName is used to define HTML form input field parameter name.
+	FormTokenName = "csrftoken"
+	// HeaderTokenName is used to define the header field name which can contain the token.
+	HeaderTokenName = "Csrf-Token"
```
Comment (Ompluscator) [inline application/service.go]: Expected standard name would be "X-CSRF-TOKEN" (like in this case: https://laravel.com/docs/5.8/csrf#csrf-x-csrf-token).

File: application/service.go [hunk 2]
```diff
@@ -91,12 +93,16 @@ func (s *ServiceImpl) IsValid(request *web.Request) bool {
 		return true
 	}
 
-	formToken, err := request.Form1(TokenName)
+	requestToken, err := request.Form1(FormTokenName)
 	if err != nil {
-		return false
+		headerCsrfToken := request.Request().Header.Get(HeaderTokenName)
```
Comment (Ompluscator) [inline application/service.go]: To be honest, I don't like this approach - it feels a little bit magical to put whole this logic into single method. Main idea of the method is to validate form POST parameter - so I would suggest to expose two methods, one for validating header parameter, second to validates form POST parameter.

**Comment by Ompluscator** [comment]
Also, check for the merge conflicts :)

File: application/service.go (hunk 1 above)
Comment (phrix32) [reply application/service.go]: Actually _X-_ prefix is deprecated (see [RFC6648 3.3](https://tools.ietf.org/html/rfc6648#section-3)

File: application/service.go (hunk 2 above)
Comment (phrix32) [reply application/service.go]: You‘re right. I‘ll separate these two checks from each other.

**Comment by phrix32** [comment]
resolved in #4

**Pull request closed without merging**